src/js/modules/trade/trade.js

```diff
@@ -0,0 +1,383 @@
+    }
+
+    handlePlaceOrder(BuySell) {
+        this.currentOrder.Orders = [];
+        this.currentOrder.BuySell = BuySell;
+
+        const obj = { selectedAccount: this.state.selectedAccount, ...this.props };
+        if (this.state.takeProfitOpen) {
+            // Setup related order
+            const order = queries.getRelatedOrder('Limit', this.takeProfitPrice, this.currentOrder);
+            this.currentOrder.Orders.push(order);
+        }
+        if (this.state.stopLossOpen) {
+            // Setup another related order
+            const order = queries.getRelatedOrder(this.stopLossOrderType, this.stopLossPrice, this.currentOrder);
+            order.StopLimitPrice = this.stopLossPrice;
+            this.currentOrder.Orders.push(order);
+        }
+        queries.postOrder(this.currentOrder, this.props, (response) => {
+            this.setState({ responseData: response });
+            ToastAndroid.show('Order placed !! ', ToastAndroid.SHORT);
+            this.props.navigation.navigate('OrderAndPosition', { ...obj });
+        });
+    }
+
+    handleOrderDurationChange(buttonIndex) {
+        this.currentOrder.OrderDuration.DurationType = orderDuration[buttonIndex];
+        this.setState({
+            updated: !this.state.updated,
+        });
+    }
+
+    handleOrderTypeChange(buttonIndex) {
+        this.currentOrder.OrderType = this.state.supportedOrderTypes[buttonIndex];
+        this.setState({ updated: !this.state.updated });
+    }
+
+    handleOrderPriceChange(text) {
+        this.currentOrder.OrderPrice = text;
+        this.setState({ updated: !this.state.updated });
+    }
+
+    handleOrderQuantityChange(text) {
+        this.currentOrder.Amount = text;
+        this.setState({ updated: !this.state.updated });
+    }
+
+    showAlert(BuySell) {
+        Alert.alert(
+            'ORDER CONFIRMATION',
+            'Do you want to place your order? ',
+            [
+                { text: 'Cancel' },
+                { text: 'OK', onPress: () => this.handlePlaceOrder(BuySell) },
+            ],
+        );
+    }
+
+    render() {
+        const { instrument, isLoading } = this.props;
+        const assetType = (instrument && instrument.AssetType === 'CfdOnStock') ? 'CFD' : instrument.AssetType;
+        const DisplayAndFormat = this.state.instrumentInfo ? this.state.instrumentInfo.DisplayAndFormat : null;
+        const PriceInfo = this.state.instrumentInfo ? this.state.instrumentInfo.PriceInfo : null;
+        const PriceInfoDetails = this.state.instrumentInfo ? this.state.instrumentInfo.PriceInfoDetails : null;
+        const InstrumentPriceDetails = this.state.instrumentInfo ? this.state.instrumentInfo.InstrumentPriceDetails : null;
+        const Quote = this.state.instrumentInfo ? this.state.instrumentInfo.Quote : null;
+        const marketStatus = (InstrumentPriceDetails && InstrumentPriceDetails.IsMarketOpen) ? 'Market Open' : 'Market Closed';
```

Comment:
values 'Market Open','Market closed', ; these can be put in constants file by making object.
for eg:  export const TRADE_MARKET: {
'MARKET_OPEN': 'Market Open;,
}